src/visit.ts

```diff
@@ -563,6 +668,94 @@ export function visitEachChild<T extends FunctionlessNode>(
     } else {
       return new ArrayBinding(bindings as ArrayBinding["bindings"]) as T;
     }
+  } else if (isLabelledStmt(node)) {
+    const stmt = visitor(node.stmt);
+    ensure(stmt, isStmt, "LabelledStmt's stmt must be a Stmt");
+    return new LabelledStmt(node.label, stmt) as T;
+  } else if (isSwitchStmt(node)) {
+    const clauses = node.clauses.flatMap((clause) => {
+      const updatedClause = visitor(clause);
+      if (Array.isArray(updatedClause)) {
+        ensureItemOf(
+          updatedClause,
+          isSwitchClause,
+          "must be a CaseClause or DefaultClause"
+        );
+        return updatedClause;
+      } else {
+        ensure(
+          updatedClause,
+          isSwitchClause,
+          "must be a CaseClause or DefaultClause"
+        );
+        return [updatedClause];
+      }
+    });
+
+    const defaultClauses = clauses.filter(isDefaultClause);
+    if (defaultClauses.length === 1) {
+      if (!isDefaultClause(clauses[clauses.length - 1])) {
+        throw new Error(`only the last SwitchClause can be a DefaultClause`);
+      }
+    } else if (defaultClauses.length > 1) {
+      throw new Error(
+        `there must be 0 or 1 DefaultClauses in a single SwitchStmt, but found ${defaultClauses.length}`
+      );
```

Comment:
SynthError?

Reply:
Not right now - we don't have SynthError for `visit`, plus these are internal errors. If we hit these errors it is entirely our fault, not the user's.

Reply:
There is a `ErrorCodes.Unexpected_Error` for that purpose, you could just catch all errors in here and throw that.